packages/react/src/component-tests/IcPopoverMenu/IcPopoverMenu.cy.tsx

```diff
@@ -0,0 +1,112 @@
+
+describe("IcPopoverMenu", () => {
+  beforeEach(() => {
+    cy.injectAxe();
+  });
+
+  afterEach(() => {
+    cy.task("generateReport");
+  });
+
+  it("renders defaul", () => {
+    mount(<PopoverDropdown></PopoverDropdown>);
+    cy.checkHydrated("ic-popover-menu");
+    cy.checkA11yWithWait();
+    cy.get("ic-button").click();
+    cy.compareSnapshot({
+      name: "default_afterclick_button",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+  });
+
+  it("renders with description", () => {
+    mount(<PopoverMenuDescription></PopoverMenuDescription>);
+    cy.checkA11yWithWait();
+    cy.get("ic-button").click();
+    cy.compareSnapshot({
+      name: "popovermenu_with_description",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+  });
+
+  it("should not be clickable when disabled", () => {
+    cy.spy(window.console, "log").as("spyWinConsoleLog");
+    mount(<DisabledPopoverMenu onClick={() => console.log()} />);
+    cy.checkA11yWithWait();
+    cy.checkHydrated("ic-popover-menu");
+    cy.get("ic-button").click();
+    cy.get("ic-menu-item").eq(0).click({ force: true });
+    cy.get("ic-menu-item").eq(1).click({ force: true });
+    cy.get("ic-menu-item").eq(2).click({ force: true });
+    cy.compareSnapshot({
+      name: "disabled_dropdown",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+    cy.get(WIN_CONSOLE_SPY).should(NOT_BE_CALLED_ONCE);
+  });
+
+  it("renders with button variations", () => {
+    mount(<PopoverMenuWithVariants></PopoverMenuWithVariants>);
+    cy.checkA11yWithWait();
+    cy.get("ic-button").click();
+    cy.compareSnapshot({
+      name: "popoverMenu_with_variants",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+  });
+
+  it("render PopoverMenu With button With menu group", () => {
+    mount(<PopoverWithMenuGroups></PopoverWithMenuGroups>);
+    cy.checkA11yWithWait();
+    cy.get("ic-button").click();
+    cy.compareSnapshot({
+      name: "popoverMenu_with_menu_group",
+      testThreshold: setThresholdBasedOnEnv(0.02),
+    });
+  });
+
+  it("should emit icPopoverClosed when document body is clicked", () => {
+    mount(<PopoverDropdown />);
+    cy.checkA11yWithWait();
+    cy.get("ic-popover-menu").invoke(
+      "on",
+      "icPopoverClosed",
+      cy.stub().as("icPopoverClosed")
+    );
+    cy.get("ic-button").click();
+    cy.get("body").click("bottomRight");
+    cy.get("@icPopoverClosed").should(HAVE_BEEN_CALLED_ONCE);
+  });
+
+  it("should emit triggerPopoverMenuInstance when submenu trigger button is clicked", () => {
+    mount(<PopoverDropdown />);
+    cy.checkA11yWithWait();
+    cy.get("ic-popover-menu").invoke(
+      "on",
+      "triggerPopoverMenuInstance",
+      cy.stub().as("triggerPopoverMenuInstance")
+    );
+
+    cy.get("ic-button").click();
+
+    cy.get("#submenu-trigger-actions").click();
+
+    cy.get("@triggerPopoverMenuInstance").should(HAVE_BEEN_CALLED_ONCE);
+  });
+});
```

Comment:
Please can you add another test very similar to this for `handleMenuItemClick`? It should be exactly the same, but you need to add an id to a regular IcMenuItem in the data file.